sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -493,6 +496,93 @@ public String getId() {
         return Utils.getSimpleObjectMapper().convertValue(object, classType);
     }
 
+    /**
+     * Executes the transactional batch.
+     *
+     * @param transactionalBatch Batch having list of operation and partition key which will be executed by this container.
+     *
+     * @return A Mono response which contains details of execution of the transactional batch.
+     * <p>
+     * If the transactional batch executes successfully, the value returned by {@link
+     * TransactionalBatchResponse#getResponseStatus} on the response returned will be set to 200}.
+     * <p>
+     * If an operation within the transactional batch fails during execution, no changes from the batch will be
+     * committed and the status of the failing operation is made available by {@link
+     * TransactionalBatchResponse#getResponseStatus}. To obtain information about the operations that failed, the
+     * response can be enumerated. This returns {@link TransactionalBatchOperationResult} instances corresponding to
+     * each operation in the transactional batch in the order they were added to the transactional batch. For a result
+     * corresponding to an operation within the transactional batch, use
+     * {@link TransactionalBatchOperationResult#getResponseStatus}
+     * to access the status of the operation. If the operation was not executed or it was aborted due to the failure of
+     * another operation within the transactional batch, the value of this field will be 424;
+     * for the operation that caused the batch to abort, the value of this field
+     * will indicate the cause of failure.
+     * <p>
+     * The value returned by {@link TransactionalBatchResponse#getResponseStatus} on the response returned may also have
+     * values such as 500 in case of server errors and 429.
+     * <p>
+     * Use {@link TransactionalBatchResponse#isSuccessStatusCode} on the response returned to ensure that the
+     * transactional batch succeeded.
+     */
+    @Beta(Beta.SinceVersion.V4_7_0)
+    public Mono<TransactionalBatchResponse> executeTransactionalBatch(TransactionalBatch transactionalBatch) {
```

Comment:
container.createItem() will always either succeed and returns a CosmosItemResponse or will throw an exception on failure.

however here it seems we are allowing returning a TransactionalBatchResponse even if the whole batch fails (you provided error methods such as (`TransactionalBatchResponse.getErrorMessage()`). 

this is not consistent with our existing apis where on failure we throw.

shouldn't we follow the same pattern here?

/CC @FabianMeiswinkel @kirankumarkolli

Reply:
This is something backend is doing here, in case of a user error like conflict or not found, backend returns overall status as 207 with individual status code having the failure status codes.

So if we are to throw an exception, that has to be triggered by SDK. And if we do throw an exception by SDK, results of individual operations will be lost and user won't be able to determine that which particular operation failed as all they will get is a cosmos exception.

Reply:
what about network error? are we differentiating network error when the whole batch fails?

Reply:
Yeah we can do that. Will make the change. Thanks.